common-lib/test/cmr/common/test/generics.clj

```diff
@@ -0,0 +1,38 @@
+(ns cmr.common.test.generics
+  (:require
+   [cmr.common.generics :as gconfig]
+   [clojure.string :as string]
+   [clojure.test :refer :all]))
+
+(deftest read-generic-doc-file-test
+  (testing "Reads the markdown file for the given generic if the generic is not in the system or it is the wrong version, return empty string"
+    (is (= true (string/includes? (gconfig/read-generic-doc-file "ingest" :grid "0.0.1") "Create / Update a Grid")))
+    (is (pos? (count (gconfig/read-generic-doc-file "ingest" :grid "0.0.1"))))
+    (is (= true (string/includes? (gconfig/read-generic-doc-file "search" :grid "0.0.1") "Searching for grids")))
+    (is (pos? (count (gconfig/read-generic-doc-file "search" :grid "0.0.1"))))
+    (is (= 0 (count (gconfig/read-generic-doc-file "ingest" :notGenericInSystem "0.0.1"))))
+    (is (= 0 (count (gconfig/read-generic-doc-file "ingest" :grid "8.23.1"))))
+    (is (= 0 (count (gconfig/read-generic-doc-file "non-real-docs" :grid "0.0.1"))))))
+
+;;This function is concatenating all of the generic documentation into one string to be passed into the api docs
+(deftest all-generic-docs-test
+  (testing "This string is ensuring that all the generics have their text concatanated together"
+    (is (= true (string/includes? (gconfig/all-generic-docs "ingest") "Grid")))
+    ))
+
+(deftest get-generics-with-documenation-test
+  (testing "This should return a list of the names of the documents currently in the system, ensure that grid is in the system"
+    (is (= "grid" (some #{"grid"} (gconfig/get-generics-with-documenation (gconfig/all-generic-docs "ingest")))))))
+
+(deftest table-of-contents-html-test
```

Comment:
can this be cleaned up by writing it with are3?

Reply:
`### <a name="grid"></a> Grid` contains all the parts you need for an anchor tag and are the same across all sections of the documents

- the pounds tell you what hx to use, in this case, h3,
- the name of the a tag tells you the href to use in the table of context a tag, in this case `#grid`
- the Text after the a tag tells you the human readable table of contents text